src/vm/arm64/crthelpers.asm

```diff
@@ -12,85 +12,13 @@
 
     TEXTAREA
 
-; Calls to JIT_MemSet is emitted by jit for initialization of large structs. 
-; We need to provide our own implementation of memset instead of using the ones in crt because crt implementation does not gurantee 
-; that aligned 8/4/2 - byte memory will be written atomically. This is required because members in a struct can be read atomically 
-; and their values should be written atomically.
-; 
-;
 ;void JIT_MemSet(void *dst, int val, SIZE_T count)
 ;{
 ;    uint64_t valEx = (unsigned char)val;
 ;    valEx = valEx | valEx << 8;
 ;    valEx = valEx | valEx << 16;
 ;    valEx = valEx | valEx << 32;
 ;
-;    size_t dc_zva_size = 4ULL << DCZID_EL0.BS;
-;
-;    uint64_t use_dc_zva = (val == 0) && !DCZID_EL0.p ? count / (2 * dc_zva_size) : 0; // ~Minimum size (assumes worst case alignment)
-;
-;    // If not aligned then make it 8-byte aligned   
-;    if(((uint64_t)dst&0xf) != 0)
-;    {
-;        // Calculate alignment we can do without exceeding count
-;        // Use math to avoid introducing more unpredictable branches
-;        // Due to inherent mod in lsr, ~7 is used instead of ~0 to handle count == 0
-;        // Note logic will fail is count >= (1 << 61).  But this exceeds max physical memory for arm64
-;        uint8_t align = (dst & 0x7) & (~uint64_t(7) >> (countLeadingZeros(count) mod 64))
-;
-;        if(align&0x1)
-;        {
-;            *(unit8_t*)dst = (unit8_t)valEx;
-;            dst = (unit8_t*)dst + 1;
-;            count-=1;
-;        }
-;
-;        if(align&0x2)
-;        {
-;            *(unit16_t*)dst = (unit16_t)valEx;
-;            dst = (unit16_t*)dst + 1;
-;            count-=2;
-;        }
-;
-;        if(align&0x4)
-;        {
-;            *(unit32_t*)dst = (unit32_t)valEx;
-;            dst = (unit32_t*)dst + 1;
-;            count-=4;
-;        }
-;    }
-;
-;    if(use_dc_zva)
-;    {
-;        // If not aligned then make it aligned to dc_zva_size
-;        if(dst&0x8)
-;        {
-;            *(uint64_t*)dst = (uint64_t)valEx;
-;            dst = (uint64_t*)dst + 1;
-;            count-=8;
-;        }
-;
-;        while(dst & (dc_zva_size - 1))
-;        {
-;            *(uint64_t*)dst = valEx;
-;            dst = (uint64_t*)dst + 1;
-;            *(uint64_t*)dst = valEx;
-;            dst = (uint64_t*)dst + 1;
-;            count-=16;
-;        }
-;
-;        count -= dc_zva_size;
-;
-;        while(count >= 0)
-;        {
-;            dc_zva(dst);
-;            dst = (uint8_t*)dst + dc_zva_size;
-;            count-=dc_zva_size;
-;        }
-;
-;        count += dc_zva_size;
-;    }
-;
```

Comment:
Remove attempts to align `dst` and attempts to use `dc zva`.  

Based on comments elsewhere the distribution of use cases tends to be biased to toward small objects.  2013 experiments showed 95% smaller than 64 bytes.  Use of `dc zva` add complexity and a series of branches which may not be predictable.  It could not be used for A53, A57 or Centriq until the length was 128 bytes in this design.

Removing `dc zva` is probably a win.  Adding it was probably a mistake on my part.